proposals/0200-raw-string-escaping-revision-1.md

```diff
@@ -0,0 +1,454 @@
+| Escape Delimiter + `"` | Quotation Mark (U+0022) | Punctuation / Quotation Mark U+0022 |
+| Escape Delimiter + `'` | Apostrophe (U+0027) | Punctuation / Apostrophe (Single Quote) U+0027 |
+| Escape Delimiter + `u{n}` | Latin Small Letter U (U+0055), Left Curly Bracket (U+007B), Right Curly Bracket (U+007D) | An arbitrary Unicode scalar, where *n* is a 1–8 digit hexadecimal number with a value equal to a valid Unicode code point |
+| Escape Delimiter + `(...)` | Left Parenthesis (U+0028), Right Parenthesis (U+0029) | An interpolated expression |
+
+
+## Examples
+
+Consider the text `\#1`:
+
+```
+// What you type today
+"\\#1" // escape delimiter + backslash + # + 1
+
+// What you type using the new system
+"\\#1" // the same
+
+// What you type with a single pound string delimiter
+#"\#\#1"# // escape delimiter + backslash + # + 1
+
+// Or you can adjust the string delimiter for a raw string:
+##"\#1"## // backslash + # + 1, no escape sequence
+```
+
+Adjusting string delimiters allows you to eliminate escape sequences to present text as intended for use:
+
+```
+#"c:\windows\system32"# // vs. "c:\\windows\\system32"
+#"\d{3) \d{3} \d{4}"# // vs "\\d{3) \\d{3} \\d{4}"
+
+#"a string with "double quotes" in it"#
+
+##"a string that needs "# in it"##
+
+#"""
+    a string with 
+    """
+    in it
+    """#
+```
+
+The following example terminates with backslash-r-backslash-n:
+
+```
+#"a raw string containing \r\n"# 
+// vs "a raw string containing \\r\\n"
+```
+
+The same behavior is extended to multi-line strings:
+
+```
+#"""
+    a raw string containing \r\n
+    """#
+```
+
+Custom-delimited strings allow you to incorporate already-escaped text. For example, you can paste static data without having to worry about re-escaping a JSON message
+
+```
+#"""
+	[
+		{
+			"id": "12345",
+			"title": "A title that \"contains\" \\\""
+		}
+	]
+	"""#
+```
+
+Without custom delimiters, Swift would silently unescape this content, yielding an invalid JSON message. Even if you did remember to escape, this process would be error-prone and difficult to maintain.
+
+## Errors
+
+The compiler errors when an escape delimiter is followed by an unrecognized value to complete an escape sequence. For example, using one-`#`-delimited strings:
+
+```
+#"printf("%s\n", value_string)"# // no error, no escape sequence
+#"printf("%s\#n", value_string)"# // newline escape sequence
+#"printf("%s\#x\#n", value_string)"# // error: escape delimiter + x. 
+```
+
+The last example can introduce a fixit by adding another `#` to either side of the string so `\#` is no longer the escape delimiter. However, this eliminates the subsequent line feed (a valid escape sequence) that follows unless that, too, is appropriately updated.
+
+There are also wrong ways to add interpolated text. These examples are both errors. The escape delimiter in each case (respectively `\` and `\#`) is followed by `#`, forming an invalid escape sequences:
```

Comment:
Should be either `forming an invalid escape sequence` or `forming invalid escape sequences`.